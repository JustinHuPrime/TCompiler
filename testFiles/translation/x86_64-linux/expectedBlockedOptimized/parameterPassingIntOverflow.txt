testFiles/translation/x86_64-linux/input/parameterPassingIntOverflow.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(5,
    MOVE(TEMP(temp6, 1, 1, GP), REG(rdi, 1)),
    MOVE(TEMP(temp7, 2, 2, GP), REG(rsi, 2)),
    MOVE(TEMP(temp8, 4, 4, GP), REG(rdx, 4)),
    MOVE(TEMP(temp9, 8, 8, GP), REG(rcx, 8)),
    MOVE(TEMP(temp10, 1, 1, GP), REG(r8, 1)),
    MOVE(TEMP(temp11, 1, 1, GP), REG(r9, 1)),
    STK_LOAD(TEMP(temp12, 2, 2, GP), CONSTANT(8, LONG(8))),
    JUMP(LOCAL(3)),
  ),
  BLOCK(3,
    MOVE(TEMP(temp15, 8, 8, GP), GLOBAL(_T3foo3bar)),
    JUMP(LOCAL(13)),
  ),
  BLOCK(13,
    MOVE(TEMP(temp17, 1, 1, GP), TEMP(temp6, 1, 1, GP)),
    JUMP(LOCAL(16)),
  ),
  BLOCK(16,
    MOVE(TEMP(temp19, 2, 2, GP), TEMP(temp7, 2, 2, GP)),
    JUMP(LOCAL(18)),
  ),
  BLOCK(18,
    MOVE(TEMP(temp21, 4, 4, GP), TEMP(temp8, 4, 4, GP)),
    JUMP(LOCAL(20)),
  ),
  BLOCK(20,
    MOVE(TEMP(temp23, 8, 8, GP), TEMP(temp9, 8, 8, GP)),
    JUMP(LOCAL(22)),
  ),
  BLOCK(22,
    MOVE(TEMP(temp25, 1, 1, GP), TEMP(temp10, 1, 1, GP)),
    JUMP(LOCAL(24)),
  ),
  BLOCK(24,
    MOVE(TEMP(temp27, 1, 1, GP), TEMP(temp11, 1, 1, GP)),
    JUMP(LOCAL(26)),
  ),
  BLOCK(26,
    MOVE(TEMP(temp28, 2, 2, GP), TEMP(temp12, 2, 2, GP)),
    JUMP(LOCAL(14)),
  ),
  BLOCK(14,
    SUB(REG(rsp, 8), REG(rsp, 8), CONSTANT(8, LONG(16))),
    MOVE(REG(rdi, 1), TEMP(temp17, 1, 1, GP)),
    MOVE(REG(rsi, 2), TEMP(temp19, 2, 2, GP)),
    MOVE(REG(rdx, 4), TEMP(temp21, 4, 4, GP)),
    MOVE(REG(rcx, 8), TEMP(temp23, 8, 8, GP)),
    MOVE(REG(r8, 1), TEMP(temp25, 1, 1, GP)),
    MOVE(REG(r9, 1), TEMP(temp27, 1, 1, GP)),
    STK_STORE(CONSTANT(8, LONG(0)), TEMP(temp28, 2, 2, GP)),
    CALL(TEMP(temp15, 8, 8, GP)),
    ADD(REG(rsp, 8), REG(rsp, 8), CONSTANT(8, LONG(16))),
    RETURN(),
  ),
)
