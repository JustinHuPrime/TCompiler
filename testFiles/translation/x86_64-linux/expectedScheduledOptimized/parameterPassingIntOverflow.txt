testFiles/translation/x86_64-linux/input/parameterPassingIntOverflow.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(0,
    MOVE(TEMP(temp6, 1, 1, GP), REG(rdi, 1)),
    MOVE(TEMP(temp7, 2, 2, GP), REG(rsi, 2)),
    MOVE(TEMP(temp8, 4, 4, GP), REG(rdx, 4)),
    MOVE(TEMP(temp9, 8, 8, GP), REG(rcx, 8)),
    MOVE(TEMP(temp10, 1, 1, GP), REG(r8, 1)),
    MOVE(TEMP(temp11, 1, 1, GP), REG(r9, 1)),
    STK_LOAD(TEMP(temp12, 2, 2, GP), CONSTANT(8, LONG(8))),
    MOVE(TEMP(temp15, 8, 8, GP), GLOBAL(_T3foo3bar)),
    MOVE(TEMP(temp17, 1, 1, GP), TEMP(temp6, 1, 1, GP)),
    MOVE(TEMP(temp19, 2, 2, GP), TEMP(temp7, 2, 2, GP)),
    MOVE(TEMP(temp21, 4, 4, GP), TEMP(temp8, 4, 4, GP)),
    MOVE(TEMP(temp23, 8, 8, GP), TEMP(temp9, 8, 8, GP)),
    MOVE(TEMP(temp25, 1, 1, GP), TEMP(temp10, 1, 1, GP)),
    MOVE(TEMP(temp27, 1, 1, GP), TEMP(temp11, 1, 1, GP)),
    MOVE(TEMP(temp28, 2, 2, GP), TEMP(temp12, 2, 2, GP)),
    SUB(REG(rsp, 8), REG(rsp, 8), CONSTANT(8, LONG(16))),
    MOVE(REG(rdi, 1), TEMP(temp17, 1, 1, GP)),
    MOVE(REG(rsi, 2), TEMP(temp19, 2, 2, GP)),
    MOVE(REG(rdx, 4), TEMP(temp21, 4, 4, GP)),
    MOVE(REG(rcx, 8), TEMP(temp23, 8, 8, GP)),
    MOVE(REG(r8, 1), TEMP(temp25, 1, 1, GP)),
    MOVE(REG(r9, 1), TEMP(temp27, 1, 1, GP)),
    STK_STORE(CONSTANT(8, LONG(0)), TEMP(temp28, 2, 2, GP)),
    CALL(TEMP(temp15, 8, 8, GP)),
    ADD(REG(rsp, 8), REG(rsp, 8), CONSTANT(8, LONG(16))),
    RETURN(),
  ),
)
