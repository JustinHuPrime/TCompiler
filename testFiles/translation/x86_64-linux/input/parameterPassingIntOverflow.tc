module foo;

void bar(ubyte a, short b, uint c, long d, char e, bool f,
         short inMemory) {
  bar(a, b, c, d, e, f,
      inMemory);
}