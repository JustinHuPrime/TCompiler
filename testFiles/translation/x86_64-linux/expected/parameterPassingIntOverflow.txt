testFiles/translation/x86_64-linux/input/parameterPassingIntOverflow.tc:
TEXT(_T3foo3bar,
  BLOCK(5,
    MOVE(TEMP(temp6, 1, 1, GP), REG(rdi)),
    MOVE(TEMP(temp7, 2, 2, GP), REG(rsi)),
    MOVE(TEMP(temp8, 4, 4, GP), REG(rdx)),
    MOVE(TEMP(temp9, 8, 8, GP), REG(rcx)),
    MOVE(TEMP(temp10, 1, 1, GP), REG(r8)),
    MOVE(TEMP(temp11, 1, 1, GP), REG(r9)),
    STK_LOAD(TEMP(temp12, 2, 2, GP), OFFSET(8)),
    JUMP(LABEL(.L3)),
  ),
  BLOCK(3,
    JUMP(LABEL(.L4)),
  ),
  BLOCK(4,
    RETURN(),
  ),
)
